crates/core/src/node/inner/in_memory_inner.rs

```diff
@@ -471,10 +473,29 @@ impl InMemoryNodeInner {
         } = self.run_l2_tx_raw(l2_tx.clone(), vm)?;
 
         if let ExecutionResult::Halt { reason } = result.result {
-            // Halt means that something went really bad with the transaction execution (in most cases invalid signature,
-            // but it could also be bootloader panic etc).
-            // In such case, we should not persist the VM data, and we should pretend that transaction never existed.
-            anyhow::bail!("Transaction HALT: {reason}");
+            let reason_clone = reason.clone();
+
+            let halt_error = tokio::runtime::Handle::current().block_on(async {
+                tokio::task::spawn_blocking(move || {
+                    // Create a new multi-threaded runtime explicitly.
+                    let rt = tokio::runtime::Builder::new_multi_thread()
+                        .enable_all()
+                        .build()
+                        .expect("Failed to create multi-threaded runtime");
+                    rt.block_on(reason_clone.to_halt_error())
+                })
+                .await
+                .expect("spawn_blocking failed")
+            });
+
+            print_execution_error(&halt_error, Some(&l2_tx));
```

Comment:
Could use some guidance here, this fails the unit tests due to runtime error. It would be ideal if we can still handle halt reasons and provide users with an informative error message here rather than just bailing with halt reason. 

Any insight on how I can handle the threading [issue](https://github.com/matter-labs/anvil-zksync/actions/runs/13466099203/job/37632154790?pr=592)? I suppose I can make the unit tests multi-threaded (e.g. `#[tokio::test(flavor = "multi_thread")]`) but that seems like a workaround.